src/SFML/Network/IpAddress.cpp

```diff
@@ -210,7 +225,16 @@ bool operator !=(const IpAddress& left, const IpAddress& right)
 ////////////////////////////////////////////////////////////
 bool operator <(const IpAddress& left, const IpAddress& right)
 {
-    return left.toInteger() < right.toInteger();
+    if (left.m_valid < right.m_valid)
+        return true;
+
+    if (right.m_valid < left.m_valid)
+        return false;
+
+    if (left.m_address < right.m_address)
+        return true;
+
+    return false;
 }
```

Comment:
Just wondering, is there a reason why you prefer this complicated case differentiation over a simple
```cpp
return std::make_pair(left.m_valid, left.m_address)
     < std::make_pair(right.m_valid, right.m_address);
```
as suggested? The latter is much more concise, clear and easier to maintain/extend.